js/src/api/format/input.js

```diff
@@ -113,7 +113,7 @@ export function inNumber10 (number) {
 
 export function inNumber16 (number) {
   if (isInstanceOf(number, BigNumber)) {
-    return inHex(number.toString(16));
+    return inHex(number.round().toString(16));
```

Comment:
This is dangerous to do in the API, since the user needs to specify the correct. When it is not a valid int, we should actually throw an exception - needs to be handled properly where called from.